ingest/anndata_.py

```diff
@@ -61,49 +55,76 @@ def generate_cluster_header(adata, clustering_name):
         else:
             msg = f"Too few dimensions for visualization in obsm \"{clustering_name}\", found {clustering_dimension}, expected 2 or 3."
             raise ValueError(msg)
-        with open(f"{clustering_name}.cluster.anndata_segment.tsv", "w") as f:
+        file_prefix = f"{study_file_id}.{clustering_name}"
+        with open(AnnDataIngestor.set_clustering_filename(file_prefix), "w") as f:
             f.write('\t'.join(headers) + '\n')
 
     @staticmethod
-    def generate_cluster_type_declaration(adata, clustering_name):
+    def generate_cluster_type_declaration(adata, clustering_name, study_file_id):
         """
         Based on clustering dimensions, write clustering TYPE line to file
         """
         clustering_dimension = adata.obsm[clustering_name].shape[1]
         types = ["TYPE", *["numeric"] * clustering_dimension]
-        with open(f"{clustering_name}.cluster.anndata_segment.tsv", "a") as f:
+        file_prefix = f"{study_file_id}.{clustering_name}"
+        with open(AnnDataIngestor.set_clustering_filename(file_prefix), "a") as f:
             f.write('\t'.join(types) + '\n')
 
     @staticmethod
-    def generate_cluster_body(adata, clustering_name):
+    def generate_cluster_body(adata, clustering_name, study_file_id):
         """
         Append clustering data to clustering file
         """
         cluster_cells = pd.DataFrame(adata.obs_names)
         cluster_body = pd.concat(
             [cluster_cells, pd.DataFrame(adata.obsm[clustering_name])], axis=1
         )
+        file_prefix = f"{study_file_id}.{clustering_name}"
         pd.DataFrame(cluster_body).to_csv(
-            AnnDataIngestor.set_output_filename(clustering_name),
+            AnnDataIngestor.set_clustering_filename(file_prefix),
             sep="\t",
             mode="a",
             header=None,
             index=False,
         )
 
     @staticmethod
-    def files_to_delocalize(arguments):
-        # ToDo - check if names using obsm_keys need sanitization
-        cluster_file_names = [AnnDataIngestor.set_output_filename(name) for name in arguments["obsm_keys"]]
-        return cluster_file_names
+    def set_clustering_filename(name):
```

Comment:
Non-blocking:  Should we generalize this method to not just apply to clustering?  Seems like this could be used to access fragment output files of all types if we pass that information in.
```suggestion
    def set_output_filename(name, file_type):
        return f"{name}.{file_type}.anndata_segment.tsv"
```

Reply:
For now, the metadata filename is predictable ("h5ad_frag.metadata.tsv") and doesn't need to be in this function. I could see hard-coding the metadata name here rather than in ingest_pipeline.py but I'm not sure it gains us much. Plus, the clustering filenames need to accommodate the name of the clustering. We're currently just using the name of the slot in the h5ad, eventually we'll need to get the study owner preferred name (eg. tsne instead of X_tsne <- the default naming convention used by ScanPy for that slot)